ui/components/component-library/text/text.types.ts

```diff
@@ -111,4 +113,13 @@ export interface TextProps extends BoxProps {
    * Additional className to assign the Text component
    */
   className?: string;
+  /**
+   * Optional href prop enables rendering of an anchor 'a' tag
+   */
+  href?: string;
+  /**
+   * Optional disabled prop enablesdds styling to indicate
+   * non-clickability.
+   */
+  disabled?: boolean;
```

Comment:
I have tired it but the typescript complains.

Reply:
I'm suggestion to allow for the generalized version of `React.HTMLAttributes<HTMLElement>` so no specific to a button or anchor tag. Does that work?

Reply:
The BoxProps already extends  ```React.HTMLAttributes<HTMLElement> ``` so it should have worked without re-extending again. However, even if we do this TS still complains.
```shell
export interface ButtonBaseProps extends BoxProps, React.HTMLAttributes<HTMLElement> {
```

Reply:
Sorry for the delay in response. I'm hoping to checkout your branch tomorrow and have a look. We may need to re-think the types of the `Box` component so it can easily allow for the polymorphic prop and the different types of the root html element.